build/org.eclipse.pde.build/src/org/eclipse/pde/internal/build/site/PDEState.java

```diff
@@ -481,14 +484,215 @@ public void resolveState() {
+					"java.lang.instrument", //
+					"java.lang.invoke", //
+					"java.lang.management", //
+					"java.lang.ref", //
+					"java.lang.reflect", //
+					"java.math", //
+					"java.net", //
+					"java.nio", //
+					"java.nio.channels", //
+					"java.nio.channels.spi", //
+					"java.nio.charset", //
+					"java.nio.charset.spi", //
+					"java.nio.file", //
+					"java.nio.file.attribute", //
+					"java.nio.file.spi", //
+					"java.rmi", //
+					"java.rmi.activation", //
+					"java.rmi.dgc", //
+					"java.rmi.registry", //
+					"java.rmi.server", //
+					"java.security", //
+					"java.security.acl", //
+					"java.security.cert", //
+					"java.security.interfaces", //
+					"java.security.spec", //
+					"java.sql", //
+					"java.text", //
+					"java.text.spi", //
+					"java.time", //
+					"java.time.chrono", //
+					"java.time.format", //
+					"java.time.temporal", //
+					"java.time.zone", //
+					"java.util", //
+					"java.util.concurrent", //
+					"java.util.concurrent.atomic", //
+					"java.util.concurrent.locks", //
+					"java.util.function", //
+					"java.util.jar", //
+					"java.util.logging", //
+					"java.util.prefs", //
+					"java.util.regex", //
+					"java.util.spi", //
+					"java.util.stream", //
+					"java.util.zip"));
+
+	@SuppressWarnings("nls")
+	private static final Map<String, Integer> JDK_ID_RELEASE = Map.of( //
+			//TODO: add zero?
+			"JRE-1.1", 1, //
+			"J2SE-1.2", 2, // 
+			"J2SE-1.3", 3, //
+			"J2SE-1.4", 4, //
+			"J2SE-1.5", 5, //
+			"JavaSE-1.6", 6, //
+			"JavaSE-1.7", 7, //
+			"JavaSE-1.8", 8);
+
+	private static Integer readJavaReleaseVersion(IExecutionEnvironment environment) {
+		String id = environment.getId();
+		if (id.startsWith("JavaSE-")) { //$NON-NLS-1$
+			try {
+				return Integer.parseInt(id.substring("JavaSE-".length())); //$NON-NLS-1$
+			} catch (NumberFormatException e) { // Another EE
+			}
+		}
+		return JDK_ID_RELEASE.get(id);
+	}
+
+	/**
+	 * Returns t
+	 * @param vm the VM to query
+	 * @param environment the release EE to use or null, if the VM's release should be used
+	 */
+	private static Collection<String> querySystemPackages(IVMInstall vm, IExecutionEnvironment environment) {
+		if (!JavaRuntime.isModularJava(vm)) {
+			Collection<String> packages = new HashSet<>();
+			for (LibraryLocation libLocation : JavaRuntime.getLibraryLocations(vm)) {
+				IPath path = libLocation.getSystemLibraryPath();
+				if (path != null) {
+					try (ZipFile zip = new ZipFile(path.toFile())) {
+						zip.stream().filter(e -> !e.isDirectory()).map(ZipEntry::getName).filter(n -> !n.startsWith("META-INF/")).map(n -> { //$NON-NLS-1$
+							int i = n.lastIndexOf('/');
+							return n.substring(0, i);
+						}).distinct().map(n -> n.replace('/', '.')).forEach(packages::add);
+					} catch (IOException e) {
+						ILog.get().error("Failed to read packages in JVM library for " + vm + ", at " + path, e); //$NON-NLS-1$ //$NON-NLS-2$
+					}
+				}
+			}
+			return packages;
```

Comment:
@iloveeclipse can you tell if there is already a way to query JDT for the list of packages provided by a VMInstall?
Something internal would be fine for now.

Reply:
I honestly don't know. Look around uses of `org.eclipse.jdt.internal.compiler.util.JRTUtil` / `org.eclipse.jdt.internal.compiler.util.JrtFileSystem`.

Reply:
I believe you wanted to contribute it once, see https://github.com/eclipse-jdt/eclipse.jdt.debug/issues/175

Reply:
> I honestly don't know. Look around uses of `org.eclipse.jdt.internal.compiler.util.JRTUtil` / `org.eclipse.jdt.internal.compiler.util.JrtFileSystem`.

Sorry I forgot the empathize that this about non-modular (Java-1.8 or lower) JREs.

> I believe you wanted to contribute it once, see [eclipse-jdt/eclipse.jdt.debug#175](https://github.com/eclipse-jdt/eclipse.jdt.debug/issues/175)

Yes that's right. I develop that in parallel to this change and are about to submit a PR for this.

Reply:
> I believe you wanted to contribute it once, see [eclipse-jdt/eclipse.jdt.debug#175](https://github.com/eclipse-jdt/eclipse.jdt.debug/issues/175)

Here we go: https://github.com/eclipse-jdt/eclipse.jdt.debug/pull/294